components/mediation/extensions/org.wso2.micro.integrator.transport.handlers/src/main/java/org/wso2/micro/integrator/transport/handlers/requestprocessors/swagger/format/MIServerConfig.java

```diff
@@ -60,57 +59,66 @@ public String getHost(API api) throws AxisFault {
         if (api.getHost() != null) {
             return api.getHost();
         } else {
-            //Retrieve WSDLPrefix to retrieve host
-            //If transport is limited to https, https host will generate. Otherwise http will be generated
-            TransportInDescription transportIn = axisConfiguration.getTransportIn(
-                    api.getProtocol() == RESTConstants.PROTOCOL_HTTP_ONLY ? "http" : "https");
-
-            if (transportIn != null) {
-                // Give priority to WSDLEPRPrefix
-                if (transportIn.getParameter(SwaggerConstants.WSDL_EPR_PREFIX) != null) {
-                    String wsdlPrefixParam = (String) transportIn.getParameter(SwaggerConstants.WSDL_EPR_PREFIX).getValue();
-                    if (!wsdlPrefixParam.isEmpty()) {
-                        //WSDLEPRPrefix available
+            return getHostNameFromTransport(api.getProtocol() == RESTConstants.PROTOCOL_HTTP_ONLY ? "http" : "https");
+        }
+    }
+
+    @Override
+    public String getHost(String trasport) throws AxisFault {
+        return getHostNameFromTransport(trasport);
+    }
+
+    private String getHostNameFromTransport(String transport) throws AxisFault {
+        //Retrieve WSDLPrefix to retrieve host
+        //If transport is limited to https, https host will generate. Otherwise http will be generated
```

Comment:
This comment is not valid here, rather it should be added above line 62. Still, what the comment says and what is done are opposite.

Reply:
+1 removed